src/Cool/Nodes/NodesFolderWatcher.cpp

```diff
@@ -59,4 +60,15 @@ auto NodesFolderWatcher::update(INodesDefinitionUpdater& updater) -> bool
     return has_changed;
 }
 
+auto NodesFolderWatcher::has_initialized(std::filesystem::path const& path, bool& has_changed) -> bool
```

Comment:
Doesn't need to be a member function.
Can even be a lambda declared at the beginning of update, that captures has_changed implicitly